.circleci/config.yml

```diff
@@ -0,0 +1,234 @@
+               export GOROOT=/usr/local/opt/go/libexec &&
+               export CC=gcc &&
+               export CPATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include &&
+               cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=gcc -DSYSTEM_HOME_OVERRIDE=/Users/root/project/cmake-build  ../ &&
+               make -j 6  && 
+               cp bin/taraxad bin/taraxad-macos
+        - run: 
+           name: Execute ctest 
+           command: cd cmake-build/tests && ctest --output-on-failure
+        - store_artifacts:
+           path: /Users/distiller/project/cmake-build/bin/taraxad-macos
+        - persist_to_workspace:
+           root: cmake-build
+           paths: [ bin ]
+   build-linux:
+     environment:
+        BUILD_OUTPUT_DIR: "cmake-docker-build-debug"
+        CLANG_VERSION: "1:10.0-50~exp1"
+        GO_VERSION: "1.13.7"
+        CMAKE_VERSION: "3.16.3-1ubuntu1"
+        GCC_VERSION: "4:9.3.0-1ubuntu2"
+        GFLAGS_VERSION: "2.2.2-1build1"
+        JSONCPP_VERSION: "1.7.4-3.1ubuntu2"
+        JSONRPCCPP_VERSION: "0.7.0-1build3"
+        SCRYPT_VERSION: "1.21-3"
+        GOROOT: "/usr/local/go"
+        GOPATH: "/root/.go"
+     docker:
+      - image: gcr.io/$GOOGLE_PROJECT_ID/taraxa-node/builder
+        auth:
+          username: _json_key  # default username when using a JSON key file to authenticate
+          password: $GCLOUD_SERVICE_KEY 
+     resource_class: xlarge
+     steps:
+        - checkout
+        - run:       
+           name: Checkout Submodules
+           command: |
+              git submodule sync
+              git submodule update --init --recursive --jobs 8
+        - run: 
+           command: mkdir $BUILD_OUTPUT_DIR
+        - restore_cache:
+           keys:
+            - conan-cache-v1-linux--{{ checksum "conanfile.py" }}
+            - conan-cache-v1-linux
+        - run:
+           name: Conan
+           command: conan install -if $BUILD_OUTPUT_DIR --build missing -s build_type=Release .
+        - save_cache:
+           paths:
+             - ~/.conan
+           key: conan-cache-v1-linux--{{ checksum "conanfile.py" }}
+
+       # - run: 
+       #    name: Copy /opt/taraxa/existing files to project
+       #    command: cp -rp /opt/taraxa/cmake-docker-build-debug/* /root/project/cmake-docker-build-debug/
+#       - git/checkout-with-submodules
+#       - setup_remote_docker:
+#           docker_layer_caching: true
+        - run: 
+            name: Cmake
+            command: | 
+              cd $BUILD_OUTPUT_DIR &&
+              cmake -DCMAKE_BUILD_TYPE=Release \
+              -DTARAXA_STATIC_BUILD=ON \
+              -DTARAXAD_INSTALL_DIR=./bin_install \
+              -DTARAXAD_CONF_INSTALL_DIR=./bin_install \
+              ../ 
+        - run:
+            name: make clang-format
+            command: cd $BUILD_OUTPUT_DIR && make clang-format && make clang-format-check
```

Comment:
Wouldn't it be better to add the initial test/check to be the format check? 
So we can just tell the user he should commit the code with the format already in place instead of the ci "fixing" it?